### PR TITLE
feat(github)!: remove forkMode setting

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Removes `forkMode` setting and instead relies on `forkToken` alone to indicate forking mode is active. Additionally, defines `forkToken` as an experimental setting.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — APPROVED** [review]

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -361,16 +361,11 @@ In practice, it is implemented by converting the `force` configuration into a `p
...
-
-You probably have no need for this option - it is an experimental setting for the Renovate hosted GitHub App.
-If this is set to `true` then Renovate will fork the repository into the personal space of the person owning the Personal Access Token.
-
 ## forkToken
 
-You probably don't need this option - it is an experimental setting for the Renovate hosted GitHub App.
-This should be set to a Personal Access Token (GitHub only) when `forkMode` is set to `true`.
-Renovate will use this token to fork the repository into the personal space of the person owning the Personal Access Token.
+You probably have no need for this option - it is an experimental setting developed for the Forking Renovate hosted GitHub App.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:366]:
```suggestion
You probably don't need this option - it is an experimental setting developed for the Forking Renovate hosted GitHub App.
```

I think we can simplify here. 😉

File: docs/usage/self-hosted-configuration.md [hunk 2]
```diff
@@ -361,16 +361,11 @@ In practice, it is implemented by converting the `force` configuration into a `p
...
+
+If this option is non-null then Renovate will fork the target repository into the account owning the Personal Access Token, and keep its default branch up-to-date with the target.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:368]:
```suggestion
If this option is not `null` then Renovate:

- forks the target repository into the account that owns the PAT
- keep this fork's default branch up-to-date with the target
```

I think this should be a list. 😄 

Can we find another way to say "non-`null`"? We should put our points positively. [^plain-language-guide]

[^plain-language-guide]: https://www.plainlanguage.gov/guidelines/concise/use-positive-language/

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -240,7 +240,7 @@ If so then Renovate will reflect this setting in its description and use package
 
 <!-- prettier-ignore -->
 !!! note
-    The `baseBranches` config option is not supported when `forkMode` is enabled, including in the Forking Renovate app.
+    The `baseBranches` config option is not supported when a `forkToken` has been configured, such as in the Forking Renovate app.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:243]:
```suggestion
    Do _not_ use the `baseBranches` config option when you've set a `forkToken`.
    You may need a `forkToken` when you're using the Forking Renovate app.
```

I think we should put the warning up front, and then explain _why_.

File: lib/config/options/index.ts [hunk 4]
```diff
@@ -408,23 +408,15 @@ const options: RenovateOptions[] = [
...
-    stage: 'repository',
-    type: 'boolean',
-    default: false,
-    globalOnly: true,
-  },
   {
     name: 'forkToken',
     description:
-      'Will be used on GitHub when `forkMode` is set to `true` to clone the repositories.',
+      'Configure a personal access token here to enable "fork mode".',
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:414]:
```suggestion
      'Set a personal access token here to enable "fork mode".',
```

I think `set` is easier than `configure`.

**Pull request merged**